src/cmds/qstat.c

```diff
@@ -598,55 +599,83 @@ trunc_value(char *value, int len, int wide_len, int wide)
 static void
 prt_nodes(char *nodes, int no_newl)
 {
-	int  i;
+	int  i, len;
 	char linebuf[78];
-	char *stp;
-	int   vnodeid = 0;
+	char *rest = NULL;
+	char *saveptr = NULL;
+ 	char *token = NULL;
+	char *token_cp = NULL;
+	char *subtoken = NULL;
+	char *node_name = NULL;
+	char *chunk = NULL;
+	struct sockaddr_in check_ip;
+	int ret = 0;
 
 	if ((nodes == NULL) || (*nodes == '\0'))
 		return;
 
 	i = 0;
-	stp = nodes;
-	while (*nodes != '\0') {
-		if (*stp == '[')
-			vnodeid = 1;
-		else if (*stp == ']')
-			vnodeid = 0;
-		if ((vnodeid == 0) &&
-			((*stp == '.') || (*stp == '+') || (*stp == '\0'))) {
-			/* does node fit into line? */
-			if (i + stp - nodes < 77) {
-				while (nodes < stp)
-					linebuf[i++] = *nodes++;
-			} else {
-				/* flush line and start next */
-				linebuf[i] = '\0';
-				if (no_newl)
-					printf("%s", show_nonprint_chars(linebuf));
-				else
-					printf("   %s\n", show_nonprint_chars(linebuf));
-				i = 0;
-				while (nodes < stp)
-					linebuf[i++] = *nodes++;
+	rest = strdup(nodes);
+	if (rest == NULL)
+		exit_qstat("out of memory");
+	/* The exec_host string has the format <host1>/<T1>*<P1>[+<host2>/<T2>*<P2>+... ].
+	 * We are using '+' delimiter to find each <host1>/<T1>*<P1> string.
+	 */
+	token = strtok_r(rest, "+", &saveptr);
+	while (token != NULL) {
+		token_cp = strdup(token);
+		if (token_cp == NULL)
+			exit_qstat("out of memory");
+		/* We are using '/' delimiter to extract the <host1> value
+		 * from <host1>/<T1>*<P1> string. We use the <host1> to identify if
+		 * the node is created using IP address.
+		 */
+		subtoken = strtok(token, "/");
+		chunk = token_cp + strlen(subtoken);
+		ret = inet_pton(AF_INET, subtoken, &(check_ip.sin_addr));
+		if (ret == 1) {
+			/* node name is an IP address */
+			pbs_asprintf(&node_name, "%s%s", subtoken, chunk);
+		} else {
+			/* Node name is not an IP address */
+			pbs_asprintf(&node_name, "%s%s", strtok(subtoken, "."), chunk);
+		}
+		len = strlen(node_name);
+		if (i + len < 77) {
```

Comment:
As Mike suggested, please add the reason for 77.

Reply:
@sujatapatnaik52 @mike0042 we were using 77 in this function originally. So I went with it.

Reply:
I find it strange that the variable "i" was initialized to zero and hasn't changed to this point. Perhaps a remnant of some old code that never got properly cleaned up? I suggest we remove it here.

Please use the following:
```suggestion
		if (len < PBS_MAXNODENAME) {
```
Also, change the declaration of linebuf to be of size PBS_MAXNODENAME + 1.

Reply:
We may still need "i" here, but do please make the PBS_MAXNODENAME change.

Reply:
@hirenvadalia explained the value 77 comes from CHAR_LINE_LIMIT which has the value 78. It is not related to PBS_MAXNODENAME. I will change 77 and linebuf[78] to use CHAR_LINE_LIMIT - 1 and CHAR_LINE_LIMIT  respectively.